keystore/kms/key_encryptor.go

```diff
@@ -0,0 +1,70 @@
+package kms
+
+import (
+	"context"
+	"errors"
+	"github.com/cossacklabs/acra/keystore"
+)
+
+// KmsEncryptor errors
+var (
+	ErrMissingKeyPurpose     = errors.New("key purpose is required for keyID creating")
+	ErrUnsupportedKeyPurpose = errors.New("unsupported KeyPurpose option provided")
+	ErrEmptyClientIDProvided = errors.New("empty clientID in key context")
+	ErrEmptyZoneIDProvided   = errors.New("empty zoneID in key context")
+)
+
+// KeyEncryptor implementation of KMS keystore.KeyEncryptor
+type KeyEncryptor struct {
+	kmsEncryptor Encryptor
+}
+
+// NewKeyEncryptor create new KeyEncryptor
+func NewKeyEncryptor(kmsEncryptor Encryptor) *KeyEncryptor {
+	return &KeyEncryptor{
+		kmsEncryptor,
+	}
+}
+
+// Encrypt return encrypted key using KMS encryptor and context.
+func (encryptor *KeyEncryptor) Encrypt(ctx context.Context, key []byte, keyContext keystore.KeyContext) ([]byte, error) {
+	keyID, err := getKeyIDFromContext(keyContext)
+	if err != nil {
+		// TODO: add logging
+		return nil, err
```

Comment:
we can try to get logger from the context via `logging.GetLoggerFromContext` function